kolibri/core/kolibri_plugin.py

```diff
@@ -2,14 +2,160 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
-from kolibri.core.webpack.hooks import FrontEndCoreAssetHook
+from django.conf import settings
+from django.contrib.staticfiles.templatetags.staticfiles import static
+from django.core.urlresolvers import get_resolver
+from django.core.urlresolvers import reverse
+from django.template.loader import render_to_string
+from django.utils.html import mark_safe
+from django.utils.translation import get_language
+from django.utils.translation import get_language_bidi
+from django.utils.translation import get_language_info
+from django_js_reverse.core import _safe_json
+from django_js_reverse.core import generate_json
+from django_js_reverse.rjsmin import jsmin
+
+import kolibri
+from kolibri.core.content.utils.paths import get_content_storage_url
+from kolibri.core.device.models import ContentCacheKey
+from kolibri.core.oidc_provider_hook import OIDCProviderHook
+from kolibri.core.theme_hook import ThemeHook
 from kolibri.core.webpack.hooks import WebpackBundleHook
+from kolibri.utils import i18n
+from kolibri.utils.conf import OPTIONS
 
 
-class FrontEndCoreAppAssetHook(FrontEndCoreAssetHook):
+class FrontEndCoreAppAssetHook(WebpackBundleHook):
     bundle_id = "default_frontend"
 
+    def url_tag(self):
+        # Modified from:
+        # https://github.com/ierror/django-js-reverse/blob/master/django_js_reverse/core.py#L101
+        js_name = "window.kolibriPluginDataGlobal['{bundle}'].urls".format(
+            bundle=self.unique_id
+        )
+        default_urlresolver = get_resolver(None)
+
+        data = generate_json(default_urlresolver)
+
+        js = render_to_string(
+            "django_js_reverse/urls_js.tpl",
+            {"data": _safe_json(data), "js_name": "__placeholder__"},
+            # For some reason the js_name gets escaped going into the template
+            # so this was the easiest way to inject it.
+        ).replace("__placeholder__", js_name)
+        return [
+            mark_safe(
+                """<script type="text/javascript">"""
+                + jsmin(js)
+                + """
+            {js_name}.__staticUrl = '{static_url}';
+            {js_name}.__mediaUrl = '{media_url}';
+            {js_name}.__contentUrl = '{content_url}';
```

Comment:
I'm a little confused about the relationship between this and pluginData().urls

Reply:
These lines are adding additional urls to the exported URL object that is generated by the `django-js-reverse` code that we are leveraging above - it doesn't export a static, media, or content url, so we add those in specifically.